### PR TITLE
Closes #4: Introducing Maven Repositories operations

### LINKED ISSUE #4: Built the Repository Scenario
Add the possibility do add new repositories

### PR DESCRIPTION
**Pull request by ricardozanini**

### CODE REVIEW

File: nexus/repositories_maven_test.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+		},
+		Maven: Maven{
+			LayoutPolicy:  LayoutPolicyStrict,
+			VersionPolicy: VersionPolicyRelease,
+		},
+	}
+	err := client.MavenProxyRepositoryService.AddRepository(repository)
+	assert.NoError(t, err)
+	assertRemote(t, func() error {
+		// on remote testing we can check if the repository was correctly interserted
```
Comment (LCaparelli) [inline nexus/repositories_maven_test.go:236]: Typo on "interserted"?

File: nexus/repositories_maven_test.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+		assert.Equal(t, getPasswordForTest(), p)
+		_, err := w.Write([]byte(allRepositories))
+		assert.NoError(t, err)
+	}))
+	defer teardown()
+	client := newClientForTest(server)
+	repos, err := client.MavenProxyRepositoryService.ListAll()
+	assert.NoError(t, err)
+	assert.NotEmpty(t, repos)
+	assert.Len(t, repos, 4)
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply nexus/repositories_maven_test.go:187]: LOL leftovers

File: nexus/repositories_maven_test.go (hunk 1 above)
Comment (ricardozanini) [reply nexus/repositories_maven_test.go:236]: :D

**Comment by ricardozanini** [comment]
@LCaparelli after this, I believe we can launch a v0.1 of this repo and start working on Nexus Operator to create the repos automatically.

File: nexus/repositories_maven.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	Maven         Maven         `json:"maven,omitempty"`
+	NegativeCache NegativeCache `json:"negativeCache,omitempty"`
+	Proxy         Proxy         `json:"proxy,omitempty"`
+	Storage       Storage       `json:"storage,omitempty"`
+	CleanUp       *CleanUp      `json:"cleanup,omitempty"`
+	RoutingRule   string        `json:"routingRule,omitempty"`
+}
+
+// MavenProxyRepositoryService service to handle all Maven Proxy Repositories operations
+type MavenProxyRepositoryService service
```
Comment (LCaparelli) [inline nexus/repositories_maven.go:95]: Same discussion about this matching the repository pattern more closely than a service. Naming will be bitch in this case, unfortunately... But we can discuss naming if you agree with this.

File: nexus/users.go [hunk 4]
```diff
@@ -38,8 +38,8 @@ type User struct {
 // UserService contains all operations related to the User domain
 type UserService service
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply nexus/users.go:39]: Dude, this is a simple wrapper around the Nexus REST API, which itself is a persistence storage wrapped by services to validate domain data. The level of complexity is so low, that we won't afford adding another layer just to call the another wrapper. Service -> Persistence layers are more suitable for complex architectures, nothing compared what we have in here, that is just small validation and HTTP request.

[the remark this answers is not in the document]
Comment (LCaparelli) [reply nexus/users.go:39]: Yeah, you're right, I'm over-complicating. A service is more than enough here, approving :grin:

**Pull request merged**